File: Codes/2364-count-number-of-bad-pairs.py
'''
Author: LetMeFly
Date: 2025-04-18 10:26:49
LastEditors: LetMeFly.xyz
LastEditTime: 2025-04-18 10:28:15
'''
from typing import List
from collections import defaultdict

class Solution:
    def countBadPairs(self, nums: List[int]) -> int:
        times = defaultdict(int)
        ans = len(nums) * (len(nums) - 1) // 2
        for i, v in enumerate(nums):
            ans -= times[v - i]
            times[v - i] += 1
        return ans

File: Codes/2364-count-number-of-bad-pairs.cpp
/*
 * @Author: LetMeFly
 * @Date: 2025-04-18 10:23:46
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-04-18 10:25:58
 */
#if defined(_WIN32) || defined(__APPLE__)
#include "_[1,2]toVector.h"
#endif

typedef long long ll;

class Solution {
public:
    long long countBadPairs(vector<int>& nums) {
        unordered_map<int, int> times;
        ll ans = nums.size() * (nums.size() - 1) / 2;
        for (int i = 0; i < nums.size(); i++) {
            ans -= times[nums[i] - i]++;
        }
        return ans;
    }
};